ethsigner/signer/hashicorp/src/main/java/tech/pegasys/ethsigner/signer/hashicorp/HashicorpVaultSignerFactory.java

```diff
@@ -45,47 +48,44 @@
   private static final String TIMEOUT_MESSAGE =
       "Timeout while retrieving private key from Hashicorp Vault.";
 
+  private HashicorpConfig hashicorpConfig;
+
   public TransactionSigner createSigner(final HashicorpConfig hashicorpConfig) {
-    final String response =
-        requestSecretFromVault(
-            hashicorpConfig.getSigningKeyPath(),
-            hashicorpConfig.getPort(),
-            hashicorpConfig.getHost(),
-            hashicorpConfig.getAuthFilePath(),
-            hashicorpConfig.getTimeout());
+    this.hashicorpConfig = hashicorpConfig;
+    final String response = requestSecretFromVault();
     final Credentials credentials = extractCredentialsFromJson(response);
     if (credentials.getAddress() != null) {
       LOG.debug("Successfully retrieved the credentials from the Hashicorp vault.");
     }
     return new CredentialTransactionSigner(credentials);
   }
 
-  private static String requestSecretFromVault(
-      final String signingKeyPath,
-      final int serverPort,
-      final String serverHost,
-      final Path authFilePath,
-      final long timeout) {
-    final String requestURI = HASHICORP_SECRET_ENGINE_VERSION + signingKeyPath;
+  private String requestSecretFromVault() {
+    final String requestURI = HASHICORP_SECRET_ENGINE_VERSION + hashicorpConfig.getSigningKeyPath();
 
-    return getVaultResponse(serverPort, serverHost, authFilePath, requestURI, timeout);
+    return getVaultResponse(
+        hashicorpConfig.getPort(),
+        hashicorpConfig.getHost(),
+        hashicorpConfig.getAuthFilePath(),
+        requestURI,
+        hashicorpConfig.getTimeout());
   }
 
-  private static String getVaultResponse(
+  private String getVaultResponse(
       final int serverPort,
       final String serverHost,
       final Path authFilePath,
       final String requestURI,
       final long timeout) {
     final Vertx vertx = Vertx.vertx();
```

Comment:
Seems like it will involve quite a bit of change. We can either go route of using a Singelton pattern approach which holds Vertx instance and thus visible everywhere in application OR leave it as it is for now.